docs/source/model/models-from-code.rst

```diff
@@ -0,0 +1,491 @@
+
+There are a few considerations to be aware of when evaluating whether your defined model is suitable for using the models from code approach for logging. 
+
+- **Imports**: Models from code does not capture external references for non-pip installable packages. If you have external references (see the examples below), you must define these dependencies via ``code_paths`` arguments.
+- **Execution during logging**: In order to validate that the script file that you're logging is valid, the code will be executed before being written to disk. 
+- **Requirements inference**: Packages that are imported at the top of your defined model script will be inferred as requirements if they are installable from PyPI. 
+
+.. tip::
+    If you define import statements that are never used within your script, these will still be included in the requirements listing. It is recommended to use a linter
+    that is capable of determining unused import statements while writing your implementation so that you are not including irrelevant package dependencies.
+
+Using Models From Code in a Jupyter Notebook
+--------------------------------------------
+
+`Jupyter <https://jupyter.org/>`_ (IPython Notebooks) are a very convenient way to work with AI applications and modeling in general. One slight limitation that they 
+have is in their cell-based execution model. Due to the nature of how they are defined and run, the models from code feature does not directly support defining 
+a notebook as a model. Rather, this feature requires that models are defined as Python scripts (the file extension **must end in '.py'**). 
+
+Fortunately, the folks that maintain the core kernel that Jupyter uses (`IPython <https://ipython.readthedocs.io/en/stable/interactive/magics.html>`_) have created a 
+number of magic commands that are usable within notebooks to enhance the usability of notebooks as a development environment for AI practitioners. One of the most 
+useful magic commands that can be used within any notebook environment that is based upon IPython (``Jupyter``, ``Databricks Notebooks``, etc.) is the ``%%writefile`` command.
+
+The `%%writefile <https://ipython.readthedocs.io/en/stable/interactive/magics.html#cellmagic-writefile>`_ magic command, when written as the first line of a notebook 
+cell, will capture the contents of the cell (not the entire notebook, mind you, only the current cell scope) with the exception of the magic command itself and write 
+those contents to the file that you define. 
+
+For example, running the following in a notebook:
+
+.. code-block:: none
+
+    %%writefile "./hello.py"
+
+    print("hello!")
+
+Will result in a file being created, located in the same directory as your notebook, that contains:
+
+.. code-block:: python
+
+    print("hello!")
+
+
+.. note::
+    There is an optional ``-a`` append command that can be used with the ``%%writefile`` magic command. This option will **append** the cell contents to the file 
```

Comment:
We've been burned waaaay too many times by things like this. I'm in favor of explicitly mentioning "don't do this" so that we don't get some report of a nigh-impossible-to-diagnose bug that is simply due to Jupyter's out-of-order-run-cells-multiple-times paradigm.